### PR TITLE
Allow an initial value to be returned by the data hooks

### LINKED ISSUE #190: Feature: Allow an initial value to be returned by the hooks
In NextJS the idea is that you preload the data at build time. This then gets passed to your component via props.

If I have a component that uses say useCollectionData, it would be useful to specify an initial value that the hook should return whilst the data is loading. As soon as the firebase query has loaded, this value would be ignored and replaced with the value from the firebase query.

It is currently possible to do this outside of the firebase hook but it's a little awkward and this would really simplify the code for this usecase (i.e any nextjs project)

I think it'd be a pretty simple change so happy to put up a PR if you'd be interested in accepting a feature like this.

Thanks,
Dylan.

### PR DESCRIPTION
**Pull request by dylanwatsonsoftware**
Resolves #190

As discussed, here is a PR to allow an initial value to be specified.

Questions that probably should be discussed:
- Should it be 'initialValue' or 'defaultValue'?
- Is it reasonable for it to be added as a field on the options param?

I also added a github action as I find them useful.. happy to remove if its not.
Its run doesn't seem to show up in this PR but you can find it here:
https://github.com/dylanwatsonsoftware/react-firebase-hooks/runs/4581537894?check_suite_focus=true

Merging this into the v3.0.x branch as this is the one that I'm forced to use due to a bug in firebase v9.

### CODE REVIEW

**Review by chrisbianca — COMMENTED** [review]
@dylanwatsonsoftware This looks good to me, and I'm happy with `initialValue` rather than `defaultValue`. Would you be happy to add details of the option to the README?

**Comment by dylanwatsonsoftware** [comment]
No dramas. I've had a go at adding something to the README. How's that?

**Comment by chrisbianca** [comment]
Thanks @dylanwatsonsoftware, this looks great. I'll get a release out shortly

**Comment by chrisbianca** [comment]
Release here: https://github.com/CSFrequency/react-firebase-hooks/releases/tag/v3.0.5

**Comment by dylanwatsonsoftware** [comment]
Wow that was fast! Thanks so much!

**Comment by mauriceackel** [comment]
Hi @chrisbianca I just wanted to use this feature in the current release (5.0.3) and was surprised to see that it is not part of this branch. Was there a reason for not having this as part of version 5? I have not seen any infos about this breaking change in the release notes.

I would love to see this feature in v5, so could we cherry pick those changes to the master branch?

Thanks, Maurice

File: firestore/useCollection.ts [hunk 1]
```diff
@@ -123,7 +124,7 @@ const useCollectionDataInternal = <
               transform
             )
           )
-        : undefined) as Data<T, IDField, RefField>[],
+        : options && options.initialValue) as Data<T, IDField, RefField>[],
```
Comment (mauriceackel) [inline firestore/useCollection.ts:127]: If we promote this to v5, I would suggest we change this to `options?.initialValue` as an easier way of safe navigation

**Pull request merged**